Misc test issues

domachine opened the pull request:
This PR fixes some issues found in newly created optional and informative tests

valentindey commented on lib/informativeTests/informativeTest_6_3_7.js:
```diff
@@ -44,7 +44,7 @@ export default async function informativeTest_6_3_7(doc) {
     if (reference.category !== 'self') continue
     const res = await request(reference.url, { method: 'HEAD' })
-    if (res.statusCode !== 200) {
+    if (!res.statusCode || 400 < res.statusCode) {
```
Why do we add the check `!res.statusCode` here? This additional check is not performed in `testURL` of test 6.3.6

domachine replied:
This is the case when `res.statusCode` is `0` in case of a network error which is included in `< 200` of `6.3.6`

valentindey commented on lib/informativeTests/informativeTest_6_3_7.js:
```diff
@@ -44,7 +44,7 @@ export default async function informativeTest_6_3_7(doc) {
-    if (res.statusCode !== 200) {
+    if (!res.statusCode || 400 < res.statusCode) {
       ctx.infos.push({
         instancePath: `/document/references/${i}/url`,
```
The specification requires test 6.3.7 to also check the urls at `/vulnerabilities[]/references[]/url`

tschmidtb51 reviewed (commented):
See comment.

tschmidtb51 commented on tests/all/informativeTests.js:
Please see my comment above and test both variants...

tschmidtb51 commented on tests/all/informativeTests.js:
```diff
@@ -221,12 +458,47 @@ export default [
+  },
+
+  {
+    title: 'Informative test 6.3.7 accepts 302 redirect',
```
What is the difference to test in line 436?

tschmidtb51 commented on tests/all/optionalTests.js:
```diff
@@ -44,125 +28,169 @@ export default [
-    content: sortObjectKeys(new Intl.Collator(), {
-      ...minimalDoc,
-      product_tree: {
+  ...[
```
Should add the test case with the informational advisory profile.

tschmidtb51 commented on tests/all/optionalTests.js:
```diff
@@ -44,125 +28,169 @@ export default [
+        product_tree: productTree,
+        vulnerabilities: [
+          {
+            product_status: {
+              last_affected: ['CSAFPID-9080700'],
```
Looking at that it matches because of the `product_status` not the score.

tschmidtb51 reviewed (changes requested):
Please have a look at my comments. We can discuss questions in the meeting...

tschmidtb51 commented:
> @valentindey @tschmidtb51 I think I addressed all your issues. Could you recheck, and approve if appropriate?

Sorry about the delay. Just added the review.

domachine commented:
> > @valentindey @tschmidtb51 I think I addressed all your issues. Could you recheck, and approve if appropriate?
> 
> Sorry about the delay. Just added the review.

Thank you very much for review. I think I could fix all issues.

tschmidtb51 reviewed (approved):
LGTM.
